bin/download_ncbi.py

```diff
@@ -89,166 +87,154 @@
     remove_directory(dirpath)
     mkdir(dirpath)
 
 def find_files(search_term, parent_dir):
-    """Find files in given directory using regex search term"""
+    """Find files in given directory using regex"""
     try:
         search_files = os.listdir(parent_dir)
+        return sorted([
+            os.path.join(parent_dir, f) for f in search_files
+            if re.search(search_term, f) and not f.endswith("~")
+        ])
     except FileNotFoundError:
-        print(f"WARN: {parent_dir} does not exist! Trying to fix.")
-    finally:
-        search_files = os.listdir(parent_dir)
-        found_files = sorted([os.path.join(parent_dir, search_file)
-                                for search_file in search_files
-                                if re.search(search_term, search_file) and
-                                not search_file.endswith("~")
-                            ])
-        return found_files
-
-def download_and_save_file(url, output_filepath):
-    """Download the file and save it to the user-specified path"""
-    try:
-        # Make a request to the URL
-        response = requests.get(url, stream=True)
-        response.raise_for_status()  # Raise an error for bad responses
-
-        # Open the output file in binary write mode
-        with open(output_filepath, 'wb') as output_file:
-            # Iterate over the content in chunks and write to the file
-            for chunk in response.iter_content(chunk_size=8192):
-                output_file.write(chunk)
+        print(f"⚠️  Directory does not exist: {parent_dir}")
+        return []
 
-        print(f"File downloaded and saved to: {output_filepath}")
+def download_and_save_file(url, output_filepath, max_retries=5):
+    """Download the file with retry and backoff"""
+    headers = {
+        "User-Agent": "open-source-ncbi-downloader/0.1 (+https://github.com/genomic-medicine-sweden/jasen)"
+    }
 
-    except requests.exceptions.RequestException as error_code:
-        print(f"Error downloading the file: {error_code}")
-
-def index_fasta(accn, idx_cmd, download_dir, DEVNULL):
-    """Index fasta file"""
+    for attempt in range(max_retries):
+        try:
+            response = requests.get(url, stream=True, headers=headers)
+            if response.status_code == 429:
+                wait_time = (2 ** attempt) + 1
+                print(f"🔁 Rate limited (429). Retrying in {wait_time} seconds...")
+                time.sleep(wait_time)
+                continue
+            response.raise_for_status()
+            with open(output_filepath, 'wb') as output_file:
+                for chunk in response.iter_content(chunk_size=8192):
+                    output_file.write(chunk)
+            print(f"✅ Downloaded: {output_filepath}")
+            return True
+        except requests.exceptions.RequestException as e:
+            wait_time = (2 ** attempt) + 1
+            print(f"⚠️  Attempt {attempt+1} failed: {e}")
+            if attempt < max_retries - 1:
```

Comment:
Isn't this clearer as:
```python
if attempt <= max_retries:
```

Reply:
No it's not the same, there's a for loop that uses `range()` meaning that the `attempt` ranges from `[0,1,2,3,4]` (if `max_retries` is 5:
If we wanted to use `<=` it would have to be:
```
if attempt <= max_retries -2:
```
TLDR
```
>>> def check_retries(attempt, max_retries):
...     if attempt < max_retries - 1:
...             print("A", True)
...     if attempt <= max_retries:
...             print("B", True)
...     if attempt <= max_retries -2:
...             print("C", True)
... 
>>> check_retries(3, 5)
A True
B True
C True
>>> check_retries(4, 5)
B True
>>> check_retries(5, 5)
B True
```